### ISSUE TITLE
Segmentation Fault running spiceinit with spk generated by spkwriter

### ISSUE THREAD

**Issue by volcanopele** [DOI-USGS/ISIS3#5190]
**ISIS version(s) affected**: 7.0.0

**Description**  
I am trying to create a controlled mosaic of multiple CaSSIS observations. This involves running a single-script pipeline on each observation to capture the input to jigsaw, using qnet to apply fixed control points with a basemap, then running the pipeline again with that modified control point network to create observations that can be merged together into a nice, controlled, regional mosaic. On that second run of the pipeline, the script uses jigsaw using the following:

```
jigsaw fromlist=${IMAGE_NAME}_newimage.lis cnet=${CONTROL_NETWORK} \
	onet=${IMAGE_NAME}_jigsaw_net.net model1=huber file_prefix=${IMAGE_NAME}.js \
	sigma0=1.0e-3 maxits=200 update=yes OBSERVATIONS=yes camera_angles_sigma=0.04 \
	spsolve=positions SPACECRAFT_POSITION_SIGMA=10000
```

This allows us to both adjust the pointing and the spacecraft position.

Next I generate spk and ck files based on the jigsaw processing:

```
ckwriter fromlist=${IMAGE_NAME}_goodimages.txt to=$IMAGE_NAME.bc overlap=warning
spkwriter fromlist=${IMAGE_NAME}_goodimages.txt to=$IMAGE_NAME.bsp overlap=warning
```

as you can tell by the "good images.txt" tag there, I only run jigsaw on those framelets that have more than 2 points coming out of pointreg and completely disregard the BLU framelets. With observations mode on in jigsaw this is fine. As long as at least one of the framelets for each frame is in the solution, jigsaw still runs just fine. But we need to make sure the jigsaw adjustments are applied to all framelets, so we run another instance of spiceinit on all frames, including the smithed ck and spk files:

```
ls -1 *cub > image_list.lis
getArray "image_list.lis"
for IMAGE in "${image_array[@]}"
	do
		SPICELIST=$(getkey from=$IMAGE grpname="KERNELS" keyword="InstrumentPointing")
		CKSPICE=($(echo $SPICELIST | tr "," "\n"))
		CAMERACK=${CKSPICE[2]}
		SCCK=${CKSPICE[1]}
		SMITHEDCK=$DIRECTORY/$OBSERVATION/smithedCK/${IMAGE_NAME}.bc
		if [ -f ${CONTROL_NETWORK} ]
		then
			SMITHEDSPK=$DIRECTORY/$OBSERVATION/smithedCK/${IMAGE_NAME}.bsp
			spiceinit from=$IMAGE ck="($SCCK, $CAMERACK, $SMITHEDCK)" \
				spk=$SMITHEDSPK attach=true spkpredict=true ckpredict=true
		else
			spiceinit from=$IMAGE ck="($SCCK, $CAMERACK, $SMITHEDCK)" \
				attach=true spkpredict=true ckpredict=true
		fi
	done
```

in this case, the Control_Network variable is the input to jigsaw with fixed points. I just use it to differentiate between cases where I am processing images without an input control network and when I do.

**How to reproduce**  
When I run the above, I get a segmentation fault:

./cassis_image_proc.sh: line 265:  9162 Segmentation fault: 11  spiceinit from=$IMAGE ck="($SCCK, $CAMERACK, $SMITHEDCK)" spk=$SMITHEDSPK attach=true spkpredict=true ckpredict=true

During testing without a smithedck I get the same segmentation fault. I do not get a segmentation fault with the smithed ck.

**Additional context**  
I can provide a set of test images, control networks, smithed ck and spk, and my script to help diagnose. It is also possible that this just needs an adjustment to how I am applying the smithed spk in spiceinit and this actually isn't a bug. If that's case, I could definitely use some guidance.

**Comment by volcanopele** [comment]
Tried testing with framelets that had not had spiceinit or jigsaw run on them prior to applying the smithed spk with spiceinit and I get the same segmentation fault.

**Comment by jlaura** [comment]
@volcanopele I suspect we are going to need a minimal reproducible set of data to be able to reproduce this. Would you be able to apply data from just before this step in the pipeline plus the code to be able to reproduce?

Before you do that, would you mind confirming that all of the bash variable expansion is happening as you intend it to? Specifically, is `$DIRECTORY/$OBSERVATION/smithedCK/${IMAGE_NAME}.bsp` resolving to where you expect it to?

**Comment by volcanopele** [comment]
I can confirm that bash variable expansion is going where I need it to. I also tested where I explicitly gave spiceinit the full path without variables and got the same segmentation fault.

I've created a zip file with the files necessary to reproduce and placed it on my OneDrive here: 

https://1drv.ms/u/s!AmEvxpAyXS-j6QgaNhv0ApHNF2v0?e=TydrdB

This archive contains portions of the observation directory, MY36_019767_345.
the level1C sub-directory contains the original .dat and .xml framelets so you can start from the beginning as that might be easier.
to initialize, use:

```
ls -1 *C1.xml | sed s/.xml// > MY36_019767_345_0_image_name.lis
tgocassis2isis from=\$1.xml to=\$1.cub -batchlist=MY36_019767_345_0_image_name.lis
spiceinit from=\$1.cub attach=true spkpredict=true ckpredict=true -batchlist=MY36_019767_345_0_image_name.lis
```

Then jigsaw is run:
```
jigsaw fromlist=MY36_019767_345_0_newimage.lis cnet=MY36_019767_345_0_cnetextract_net.net \
	onet=MY36_019767_345_0_jigsaw_net.net model1=huber file_prefix=MY36_019767_345_0.js \
	sigma0=1.0e-3 maxits=200 update=yes OBSERVATIONS=yes camera_angles_sigma=0.04 \
	spsolve=positions SPACECRAFT_POSITION_SIGMA=10000
```

both MY36_019767_345_0_newimage.lis, MY36_019767_345_0_goodimages.txt, and MY36_019767_345_0_cnetextract_net.net can be found in the jigsaw sub-directory.

Then we use as described above, ckwriter to generate the smithed CK and spkwriter to generate the smithed spk.

finally we run the spiceinit again on all images while inserting the smithed kernels where appropriate. 

I realized we could also use the extras attribute in spiceinit to add these in, but that makes no difference. Still get a segmentation fault.

The other sub-directories includes jigsawed which contains the files in the work directory prior to the second run of spiceinit. smithedCK contains both the smithed ck and spk. ingested contains cube files after the first run of spiceinit but before jigsaw.

**Comment by amystamile-usgs** [comment]
@volcanopele I took a look at your spk file and noticed that spkwriter is returning NaN values. This was fixed in version [ISIS 7.2.0](https://github.com/DOI-USGS/ISIS3/blob/dev/CHANGELOG.md?plain=1#L97).

I will run your jigsaw process and create a new spk under the most recent version of ISIS and confirm whether or not the previous fix is relevant to this case as well.

![image](https://github.com/DOI-USGS/ISIS3/assets/74275278/cf702dac-0193-4f4e-8af7-893fdf9c5bbf)

**Comment by amystamile-usgs** [comment]
Reran jigsaw on your set of images and reran spkwriter and confirmed that the NaNs have been fixed. I also checked that spiceinit is no longer segfaulting with the new spk file. Therefore you should be able to run your script with no segfaulting using version 7.2.0 or higher of ISIS. Please note that you must rerun jigsaw in the new version of ISIS because the NaNs are being stored in the position tables of the old cubes and spkwriter picks up these values when writing the segments. 

![image](https://github.com/DOI-USGS/ISIS3/assets/74275278/df173e3e-271a-4c3f-a325-ba34b623b733)
